Mathlib/GroupTheory/GroupAction/Amenable.lean

```diff
@@ -0,0 +1,91 @@
+/-
+Copyright (c) 2023 Matthias Uschold. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Matthias Uschold
+-/
+import Mathlib.GroupTheory.GroupAction.Defs
+import Mathlib.Data.Real.ENNReal
+import Mathlib.MeasureTheory.MeasurableSpace.Defs
+
+
+/-!
+# Amenable Monoid Actions
+
+In this file, we define amenability of a monoid action.
+A monoid action is amenable if it admits an invariant mean.
+A mean is like a probability measure, demanding
+only finite additivity instead of σ-additivity.
+
+## Main Definitions
+
+- `Mean`: defines means as finitely additive probability measures
+- `InvariantMean`: defines when a mean is invariant under a monoid action
+- `amenable`: A monoid action is amenable if there exists an invariant mean
+
+## Implementation Notes
+
+`Mean` was not implemented using MeasureTheory.ProbabilityMeasure
+because meausures are σ-additive (i.e. countably additive). In this
+setting, this would be a too strong assumption, we only want to demand
+finite additivity. Typically, the resulting measures will not be
+σ-additive.
+
+## References
+
+See [bartholdi2017amenability] for definitions
+and more information on amenable actions.
+
+-/
+
+universe u v
+variable (α : Type v) [MeasurableSpace α]
+
+/--A mean is a function from the power set of α to ENNReal that
+assigns the value 1 to the full set α, and
+is finitely additive under disjoint unions -/
+structure Mean where
+  /-- function giving the measure of a measurable subset-/
+  measureOf : {S // MeasurableSet (α := α) S} → NNReal
```

Comment:
Could you instead define it on all sets (but require `fin_add` only for measurable sets)? This will make it easier to use because one won't need to prove all the time the measurability of the objects one uses.

Reply:
Ok, I will do so